core/cautils/scaninfo.go

```diff
@@ -188,25 +187,39 @@ func (scanInfo *ScanInfo) setUseFrom() {
 	}
 }
 
-func (scanInfo *ScanInfo) setOutputFile() {
-	if scanInfo.Output == "" {
-		return
-	}
-	if scanInfo.Format == "json" {
-		if filepath.Ext(scanInfo.Output) != ".json" {
-			scanInfo.Output += ".json"
+func (scanInfo *ScanInfo) GetFormats() []string {
+	return strings.Split(scanInfo.Format, ",")
+}
+
+func (scanInfo *ScanInfo) GetOutputFiles() []string {
+	formats := scanInfo.GetFormats()
+	outputs := make([]string, 0)
+
+	for _, format := range formats {
+		if scanInfo.Output == "" {
+			outputs = append(outputs, "")
 		}
-	}
-	if scanInfo.Format == "junit" {
-		if filepath.Ext(scanInfo.Output) != ".xml" {
-			scanInfo.Output += ".xml"
+
+		output := scanInfo.Output
+
+		if format == "json" {
```

Comment:
The interface should include the "printer name" and "printer extension" functions, and we should have an object that lists the supported outputs, this way we can iterate based on name and extension so we do not need this `if` situation.